## Rubric

Instructions: DELETE this cell before you submit via a `git push` to your repo before deadline. This cell is for your reference only and is not needed in your report. 

Scoring: Out of 10 points

- Each Developing  => -2 pts
- Each Unsatisfactory/Missing => -4 pts
  - until the score is 

If students address the detailed feedback in a future checkpoint they will earn these points back


|                  | Unsatisfactory                                                                                                                                                                                                    | Developing                                                                                                                                                                                              | Proficient                                     | Excellent                                                                                                                              |
|------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------|
| Data relevance   | Did not have data relevant to their question. Or the datasets don't work together because there is no way to line them up against each other. If there are multiple datasets, most of them have this trouble | Data was only tangentially relevant to the question or a bad proxy for the question. If there are multiple datasets, some of them may be irrelevant or can't be easily combined.                       | All data sources are relevant to the question. | Multiple data sources for each aspect of the project. It's clear how the data supports the needs of the project.                         |
| Data description | Dataset or its cleaning procedures are not described. If there are multiple datasets, most have this trouble                                                                                              | Data was not fully described. If there are multiple datasets, some of them are not fully described                                                                                                      | Data was fully described                       | The details of the data descriptions and perhaps some very basic EDA also make it clear how the data supports the needs of the project. |
| Data wrangling   | Did not obtain data. They did not clean/tidy the data they obtained.  If there are multiple datasets, most have this trouble                                                                                 | Data was partially cleaned or tidied. Perhaps you struggled to verify that the data was clean because they did not present it well. If there are multiple datasets, some have this trouble | The data is cleaned and tidied.                | The data is spotless and they used tools to visualize the data cleanliness and you were convinced at first glance                      |


# COGS 108 - Data Checkpoint

## Authors


- Angelina Tran - Project administration, Experimental investigation, Software
- Ethan Dam - Data curation, Analysis, Visualization
- Katelyn Sun - Writing – review & editing, Software
- Melanie Perez - Background research, Writing – original draft, Analysis
- Nimah Mcauliffe - Analysis, Background research

## Research Question

What observable characteristics of restaurants in Philidelphia — including average rating, review volume, price level, cuisine type, geographic location, and sentiment patterns in customer reviews — are associated with higher popularity? Which of these features provide the strongest predictive signals for identifying highly popular restaurants? To what extent can these variables be used to build a model that predicts restaurant popularity and distinguishes high-performing establishments from less popular ones?

## Background and Prior Work

The restaurant industry is inherently volatile, and as time passes, maintaining a restaurant has become increasingly difficult and costly due to high inflation, surging labor costs, and shifting consumer habits. In fact, in 2025, 42% of restaurants were reported to be unprofitable.<sup>[1]</sup> Compare this to the estimated 60–70% of profitable establishments just ten years ago.<sup>[2]</sup> If a restaurant could accurately predict profitability within a specific geographical location or the viability of incorporating a certain cuisine, it could more reliably guarantee meeting an annual budget and increase its chances of longevity. Furthermore, existing restaurants can analyze the differences between high-performing and low-performing establishments to adopt best practices and improve overall performance.

Historically, the 1990s present a stark contrast: prior to the dominance of digital media, restaurant survival was lower in the first year due to significant barriers in spreading information and attracting customers. At that time, roughly 70% of restaurants survived their first year, compared to 83% today.<sup>[4]</sup> However, while first-year survival rates have generally increased, modern day restaurants face a new set of obstacles. Establishments now witness steeper drop-off rates between years one and five due to margin erosion, a phenomenon where business costs rise faster than menu prices. Ultimately, this has resulted in 10% fewer establishments surviving a five year period today than in the 1990s.<sup>[2][4]</sup>

Market conditions fluctuate tremendously even from year to year. Business trends in 2024 differed significantly from those in 2025, with 60% of restaurants reporting deteriorated conditions by the end of 2024.<sup>[3]</sup> The same volatility applies to profit margins between 2025 and 2026; margins in 2026 are predicted to grow primarily from the consumption of affluent households, despite a general pull-back in spending from middle- and lower-class households.<sup>[3]</sup> External factors like tariffs on imported food and beverage items have also posed challenges for two-thirds of operators. Average food costs are now more than 35% above pre-pandemic levels, according to the U.S. Bureau of Labor Statistics. In 2025, 82% of operators reported higher average food costs, and 68% noted that tariffs drove those costs higher.<sup>[3]</sup> 

Despite this economic pressure, some locations have thrived; for example, in 2025, Chili’s reported a 13% increase in foot traffic and a 23% increase in sales.<sup>[3]</sup> Consequently, it is becoming vital to build predictive models for restaurant performance. By analyzing the practices of profitable restaurants like Chili’s—considering variables such as average ratings, review volume, price level, cuisine type, geographic location, and sentiment patterns—operators can better mitigate financial risks and seize profit opportunities.

The role of COVID-19 in restaurant longevity research cannot be overlooked. The pandemic acted as a "forced laboratory," intensifying the search for variables that predict success and longevity. Before the pandemic, failure was largely attributed to poor location or general service quality issues that led to a lack of capital. At that time, the average lifespan of an establishment was approximately 4.5 years. Post-COVID, this figure stands significantly lower, at 1 to 3 years for those failing to adapt.<sup>[5]</sup> A key quantifier in this change is the heavy dependency on delivery and digital integration as a primary survival variable.

Current research indicates that longevity is no longer found through traditional management alone, but through the integration of multimodal data. Studies from late 2025 and early 2026, such as those from Hong Kong Polytechnic University, have introduced "Text–Image Fusion" models that analyze customer-generated content to predict failure with high precision.<sup>[6]</sup> Unlike older models that relied solely on financial spreadsheets, these AI-driven systems analyze the aesthetic quality of food photos, sentiment patterns in reviews, and even meteorological data to forecast demand. Today, a restaurant's ability to survive the five-year mark is heavily dependent on how well it optimizes for delivery only zones and utilizes dynamic pricing to capture value-conscious diners. These predictive models allow operators to transition from reactive survival to proactive growth by identifying the exact cuisine and geographic variables that align with modern consumer trends.


References

<a name="nra2026"> </a> ^ 1. National Restaurant Association. (2026). Persistent Cost Increases and Enduring Demand Will Shape the Restaurant Industry in 2026. https://www.restaurant.org/research-and-media/media/press-releases/persistent-cost-increases-and-enduring-demand-will-shape-the-restaurant-industry-in-2026/

<a name="owner2026"> </a> ^ 2. Owner.com. (2026). The Real Restaurant Failure Rate. https://www.owner.com/blog/restaurant-failure-rate

<a name="nrn2026"> </a> ^ 3. Nation’s Restaurant News. (2026). Restaurant Industry Faces Modest Growth Amid Cost Pressures in 2026. https://www.nrn.com/restaurant-insights/restaurant-industry-faces-modest-growth-amid-cost-pressures-in-2026

<a name="osu2005"> </a> ^ 4. Ohio State University. (2005). Restaurant Failure Rate Much Lower Than Assumed. https://news.osu.edu/restaurant-failure-rate-much-lower-than-assumed-study-finds/

<a name="emerald2025"> </a> ^ 5. Emerald Publishing. (2025). Survival still in dilemma: Factors influencing eateries in a COVID-prone world. https://www.emerald.com/ihr/article/39/1/50/1245382/Survival-still-in-dilemma-an-empirical-analysis-of

<a name="ssrn2025"> </a> ^ 6. SSRN / Hong Kong PolyU. (2025). Survival Through Text–Image Fusion and Joint Learning. https://papers.ssrn.com/sol3/papers.cfm?abstract_id=5285395


## Hypothesis


- We hypothesize that restaurants with higher ratings, more reviews, positive customer sentiment, and favorable locations will demonstrate significantly higher popularity. Higher popularity is measured by a composite index combining average rating and review volume. We expect review count and sentiment patterns to serve as the strongest predictors, since they reflect both customer satisfaction and online engagement. This prediction is grounded in the assumption that restaurants receiving more positive attention are more likely to attract continued customer interest.
- Our thinking behind this was that review volume and sentiment directly reflect customer engagement and satisfaction, which can increase visibility and attract more customers over time. Additionally, restaurants located in dense commercial or tourist areas are likely to experience greater foot traffic, contributing to higher popularity. Together, these factors suggest that both online engagement and geographic exposure play key roles in driving restaurant popularity.


## Data

### Data overview

Instructions: REPLACE the contents of this cell with descriptions of your actual datasets.

For each dataset include the following information
- Dataset #1
  - Dataset Name: Yelp Open Dataset - Businesses Queried to Philidelphia
  - Link to the dataset: https://business.yelp.com/data/resources/open-dataset/
  - Number of observations: 14569
  - Number of variables: 14
  - Description of the variables most relevant to this project

  The most important variables for this project include stars and review_count, which measure customer satisfaction and online engagement, respectively. Latitude and longitude capture geographic location, allowing spatial analysis of popularity patterns. Categories, attributes, and hours describe business characteristics that may influence demand, while business_id enables merging with review-based features for modeling.
  - Descriptions of any shortcomings this dataset has with repsect to the project:

   The Yelp Open Dataset has many useful observational features; however, it may bias our data to skew popularity towards online engagement instead of foot traffic which is not observed. The use of Yelp data may also create a possible selection or survivorship bias where review counts and ratings are influenced by certain demographics or platform-specific behaviors (eg., extreme opinions). These limitations shoudl be considered as we interpret predicitive results.
- Dataset #2 
  - Dataset Name: Yelp Open Dataset – Reviews Queried to Philadelphia Review Aggregates
  - Link to the dataset: https://business.yelp.com/data/resources/open-dataset/
  - Number of observations: 14569
  - Number of variables: 5
  - Description of the variables most relevant to this project

  This dataset contains aggregated review-level features at the business level for Philadelphia restaurants. Key variables include review_ct, which measures customer engagement, and avg_review_stars, which reflects overall customer sentiment. Additional features such as avg_text_len and latest_review_date capture engagement intensity and recency of activity. 
  - Descriptions of any shortcomings this dataset has with repsect to the project

  Because this dataset is derived from Yelp reviews, it inherits platform bias and does not measure offline foot traffic. Review counts may overrepresent highly engaged or extreme-opinion users, and recency may not fully capture real-time popularity trends.
-  Dataset #3 
  - Dataset Name: Yelp Open Dataset – Reviews Queried to Philadelphia Review Sample 5000
  - Link to the dataset: https://business.yelp.com/data/resources/open-dataset/
  - Number of observations: 5000
  - Number of variables: 9
  - Description of the variables most relevant to this project

  This dataset contains a random sample of 5,000 review-level observations from Philadelphia businesses. Key variables include business_id, which links reviews to businesses, stars and date, which capture rating and timing information, and text, which contains the full review content. This dataset supports exploratory sentiment analysis and text-based feature engineering for modeling popularity.

  - Descriptions of any shortcomings this dataset has with repsect to the project

  Because this dataset is a sample rather than the full population of reviews, sentiment estimates may not fully represent all businesses. Additionally, textual sentiment may reflect subjective or extreme opinions and may not directly correspond to objective popularity measures.

Each dataset deserves either a set of bullet points as above or a few sentences if you prefer that method.

If you plan to use multiple datasets, add a few sentences about how you plan to combine these datasets.

In [57]:
# Run this code every time when you're actively developing modules in .py files.  It's not needed if you aren't making modules
#
## this code is necessary for making sure that any modules we load are updated here 
## when their source code .py files are modified

%load_ext autoreload
%autoreload 2

In [58]:
# Setup code -- this only needs to be run once after cloning the repo!
# this code downloads the data from its source to the `data/00-raw/` directory
# if the data hasn't updated you don't need to do this again!

# if you don't already have these packages (you should!) uncomment this line
# %pip install requests tqdm

import sys
sys.path.append('./modules') # this tells python where to look for modules to import

import get_data # this is where we get the function we need to download data

# replace the urls and filenames in this list with your actual datafiles
# yes you can use Google drive share links or whatever
# format is a list of dictionaries; 
# each dict has keys of 
#   'url' where the resource is located
#   'filename' for the local filename where it will be stored 
datafiles = [
    { 'url': 'https://raw.githubusercontent.com/fivethirtyeight/data/refs/heads/master/airline-safety/airline-safety.csv', 'filename':'airline-safety.csv'},
    { 'url': 'https://raw.githubusercontent.com/fivethirtyeight/data/refs/heads/master/bad-drivers/bad-drivers.csv', 'filename':'bad-drivers.csv'}
]

get_data.get_raw(datafiles,destination_directory='data/00-raw/')

Overall Download Progress:  50%|█████     | 1/2 [00:00<00:00,  3.21it/s]

Successfully downloaded: airline-safety.csv


Overall Download Progress: 100%|██████████| 2/2 [00:00<00:00,  3.93it/s]


Successfully downloaded: bad-drivers.csv


### Dataset #1 

Instructions: 
1. Change the header from Dataset #1 to something more descriptive of the dataset
2. Write a few paragraphs about this dataset. Make sure to cover
   1. Describe the important metrics, what units they are in, and giv some sense of what they mean.  For example "Fasting blood glucose in units of mg glucose per deciliter of blood.  Normal values for healthy individuals range from 70 to 100 mg/dL.  Values 100-125 are prediabetic and values >125mg/dL indicate diabetes. Values <70 indicate hypoglycemia. Fasting idicates the patient hasn't eaten in the last 8 hours.  If blood glucose is >250 or <50 at any time (regardless of the time of last meal) the patient's life may be in immediate danger"
   2. If there are any major concerns with the dataset, describe them. For example "Dataset is composed of people who are serious enough about eating healthy that they voluntarily downloaded an app dedicated to tracking their eating patterns. This sample is likely biased because of that self-selection. These people own smartphones and may be healthier and may have more disposable income than the average person.  Those who voluntarily log conscientiously and for long amounts of time are also likely even more interested in health than those who download the app and only log a bit before getting tired of it"
3. Use the cell below to 
    1. load the dataset 
    2. make the dataset tidy or demonstrate that it was already tidy
    3. demonstrate the size of the dataset
    4. find out how much data is missing, where its missing, and if its missing at random or seems to have any systematic relationships in its missingness
    5. find and flag any outliers or suspicious entries
    6. clean the data or demonstrate that it was already clean.  You may choose how to deal with missingness (dropna of fillna... how='any' or 'all') and you should justify your choice in some way
    7. You will load raw data from `data/00-raw/`, you will (optionally) write intermediate stages of your work to `data/01-interim` and you will write the final fully wrangled version of your data to `data/02-processed`
4. Optionally you can also show some summary statistics for variables that you think are important to the project
5. Feel free to add more cells here if that's helpful for you


In [59]:
from IPython.display import display
import pandas as pd

# Load dataset
yelp_ph = pd.read_csv('data/01-interim/yelp_philly_business.csv')

# Create clean copy
yelp_ph_clean = yelp_ph.copy()

In [60]:
# Look at dataset size
yelp_ph_clean.shape

(14569, 14)

In [61]:
# Look at variables with empty cells
yelp_ph_clean.isna().sum()

business_id        0
name               0
address          526
city               0
state              0
postal_code        8
latitude           0
longitude          0
stars              0
review_count       0
is_open            0
attributes      1170
categories         9
hours           2784
dtype: int64

Although some variables such as `address`, `attributes`, `categories` and `hours` contain missing data, these fields are not a central part of or modeling objective. We therefore can fill and leave them for later use by filling with 'Unknown'. Removing such rows would unnecessarily reduce sample size and potentially introduce bias. Instead, we retained all businesses with complete core variables (e.g., `business_id`, `stars`, `review_count`, geographic coordinates).

To check if the missing data is systematic, we will compare the values with and without missing values to see if there is a significant difference. 

In [62]:
# Comapre stars and review counts with and without `hours`
yelp_ph_clean["hours_missing"] = yelp_ph_clean["hours"].isna().astype(int)
yelp_ph_clean.groupby("hours_missing")[["stars", "review_count"]].mean()

,stars,review_count
hours_missing,,
0,3.688969,74.730165
1,3.343930,19.951509


While average star ratings were similar between groups, businesses missing `hours` exhibited substantially lower average review counts (19 vs 75). This suggests that missingness may be systematic such as due to lower engagement rather than occurring completely at random.

We will do the same for `attributes`.

In [63]:
# Comapre stars and review counts with and without `attributes`
yelp_ph_clean["attributes_missing"] = yelp_ph_clean["attributes"].isna().astype(int)
yelp_ph_clean.groupby("attributes_missing")[["stars", "review_count"]].mean()

,stars,review_count
attributes_missing,,
0,3.653743,68.287036
1,3.271368,18.172650


Businesses missing `attributes` exhibit substantially lower average review counts (18 vs 68) and slightly lower average ratings (3.27 vs 3.65), suggesting that missingness is systematic and associated with lower engagement. This indicates that incomplete business profiles may correlate with reduced popularity.

In [64]:
# Fill empty cells with 'Unknown'
yelp_ph_clean['address'] = yelp_ph_clean['address'].fillna('Unknown')
yelp_ph_clean['attributes'] = yelp_ph_clean['attributes'].fillna('Unknown')
yelp_ph_clean['categories'] = yelp_ph_clean['categories'].fillna('Unknown')
yelp_ph_clean['hours'] = yelp_ph_clean['hours'].fillna('Unknown')

# Ensure no duplicate business IDs
yelp_ph_clean["business_id"].nunique() == len(yelp_ph_clean)

# Check duplicate rows
yelp_ph_clean.duplicated().sum()

np.int64(0)

In [65]:
print("--- DataFrame Head ---")
display(yelp_ph_clean.head()) 

print("\n--- DataFrame Columns ---")
display(list(yelp_ph_clean.columns)) 

--- DataFrame Head ---


,business_id,name,address,city,state,postal_code,latitude,longitude,stars,review_count,is_open,attributes,categories,hours,hours_missing,attributes_missing
0,MTSW4McQd7CbVtyjqoe9mw,St Honore Pastries,935 Race St,Philadelphia,PA,19107.0,39.955505,-75.155564,4.0,80,1,"{'RestaurantsDelivery': 'False', 'OutdoorSeati...","Restaurants, Food, Bubble Tea, Coffee & Tea, B...","{'Monday': '7:0-20:0', 'Tuesday': '7:0-20:0', ...",0,0
1,MUTTqe8uqyMdBl186RmNeA,Tuna Bar,205 Race St,Philadelphia,PA,19106.0,39.953949,-75.143226,4.0,245,1,"{'RestaurantsReservations': 'True', 'Restauran...","Sushi Bars, Restaurants, Japanese","{'Tuesday': '13:30-22:0', 'Wednesday': '13:30-...",0,0
2,ROeacJQwBeh05Rqg7F6TCg,BAP,1224 South St,Philadelphia,PA,19147.0,39.943223,-75.162568,4.5,205,1,"{'NoiseLevel': ""u'quiet'"", 'GoodForMeal': ""{'d...","Korean, Restaurants","{'Monday': '11:30-20:30', 'Tuesday': '11:30-20...",0,0
3,QdN72BWoyFypdGJhhI5r7g,Bar One,767 S 9th St,Philadelphia,PA,19147.0,39.939825,-75.157447,4.0,65,0,"{'Smoking': ""u'no'"", 'NoiseLevel': ""u'average'...","Cocktail Bars, Bars, Italian, Nightlife, Resta...","{'Monday': '16:0-0:0', 'Tuesday': '16:0-0:0', ...",0,0
4,Mjboz24M9NlBeiOJKLEd_Q,DeSandro on Main,4105 Main St,Philadelphia,PA,19127.0,40.022466,-75.218314,3.0,41,0,"{'RestaurantsReservations': 'False', 'Caters':...","Pizza, Restaurants, Salad, Soup","{'Tuesday': '17:0-21:30', 'Wednesday': '17:0-1...",0,0



--- DataFrame Columns ---


['business_id',
 'name',
 'address',
 'city',
 'state',
 'postal_code',
 'latitude',
 'longitude',
 'stars',
 'review_count',
 'is_open',
 'attributes',
 'categories',
 'hours',
 'hours_missing',
 'attributes_missing']

The dataset satisfies tidy data principles. Each row corresponds to one unique business (confirmed by unique business_id values), and each column represents a single business-level variable. No duplicate rows were found, and no variables are distributed across multiple columns. Therefore, the dataset is already in tidy format.

### Dataset #2 

See instructions above for Dataset #1.  Feel free to keep adding as many more datasets as you need.  Put each new dataset in its own section just like these. 

Lastly if you do have multiple datasets, add another section where you demonstrate how you will join, align, cross-reference or whatever to combine data from the different datasets

Please note that you can always keep adding more datasets in the future if these datasets you turn in for the checkpoint aren't sufficient.  The goal here is demonstrate that you can obtain and wrangle data.  You are not tied down to only use what you turn in right now.

In [66]:
yelp_review_agg = pd.read_csv('data/01-interim/yelp_philly_review_agg.csv')
yelp_review_agg.head()

yelp_review_agg.shape
yelp_review_agg.columns

Index(['business_id', 'review_ct', 'avg_review_stars', 'avg_text_len',
       'latest_review_date'],
      dtype='object')

In [67]:
yelp_review_sample = pd.read_json('data/01-interim/yelp_philly_reviews_sample_5000.jsonl', lines=True)
yelp_review_sample.head()

yelp_review_sample.shape
#yelp_review_sample.columns

(5000, 9)

## Ethics

A. Data Collection

A.1 Informed consent: If there are human subjects, have they given informed consent, where subjects affirmatively opt-in and have a clear understanding of the data used to which they consent?

This project uses publicly available Yelp API data rather than directly collecting data from human subjects. While individual reviewers have not provided consent specifically for our study, the data are publicly posted under Yelp’s platform terms. We will not contact users or collect additional personal data beyond what is publicly available.

A.2 Collection bias: Have we considered sources of bias that could be introduced during data collection and survey design and taken steps to mitigate those?

Yelp data may introduce selection bias because reviewers are self-selected and may not represent the broader population of restaurant patrons. Certain neighborhoods, cuisines, or price levels may also receive disproportionate attention. As a result, our measure of “popularity” may reflect online engagement rather than true customer volume, and we will interpret findings accordingly.

A.3 Limit PII exposure: Have we considered ways to minimize exposure of personally identifiable information (PII) for example through anonymization or not collecting information that isn't relevant for analysis?

We will not collect or store personally identifiable information (PII) such as reviewer names, profile details, or user IDs. Review text will only be used for aggregate sentiment analysis, and no individual reviewer will be identified in the dataset or final report.

A.4 Downstream bias mitigation: Have we considered ways to enable testing downstream results for biased outcomes (e.g., collecting data on protected group status like race or gender)?

Our predictive model could unintentionally favor restaurants in higher-income or high-traffic areas, reinforcing existing socioeconomic disparities. While we are not collecting protected attributes such as race or gender, we will examine whether model performance varies across neighborhoods or cuisine types to assess potential structural bias.

B. Data Storage

B.1 Data security: Do we have a plan to protect and secure data (e.g., encryption at rest and in transit, access controls on internal users and third parties, access logs, and up-to-date software)?

All collected data will be stored locally in secure, access-controlled environments and used solely for academic purposes. The data will be stored only on our personal devices and private versions repositories accessible to group members and teaching staff. Access will be limited to the project team. The data will not be shared with third parties. No sensitive personal information will be retained.

B.2 Right to be forgotten: Do we have a mechanism through which an individual can request their personal information be removed?
B.3 Data retention plan: Is there a schedule or plan to delete the data after it is no longer needed?

The dataset and derived analysis files will be retained only for the duration of the course and any related academic reporting. After the project is completed and graded, the data will be deleted from personal devices and repositories unless required for reproducibility in future coursework or for portfolios.

C. Analysis

C.1 Missing perspectives: Have we sought to address blind spots in the analysis through engagement with relevant stakeholders (e.g., checking assumptions and discussing implications with affected communities and subject matter experts)?

Our analysis relies solely on Yelp engagement data and does not capture perspectives from restaurant owners, employees, or customers who do not use Yelp. This limitation may result in incomplete representation of restaurant success and community impact.

C.2 Dataset bias: Have we examined the data for possible sources of bias and taken steps to mitigate or address these biases (e.g., stereotype perpetuation, confirmation bias, imbalanced classes, or omitted confounding variables)?

The dataset may reflect imbalances such as overrepresentation of popular neighborhoods or cuisine types. Additionally, online reviews may contain cultural or linguistic bias that could influence sentiment analysis. We will acknowledge these limitations when interpreting our results.

C.3 Honest representation: Are our visualizations, summary statistics, and reports designed to honestly represent the underlying data?

We commit to presenting visualizations and summary statistics that accurately reflect the data without exaggerating findings. We will avoid misleading scales, or overstated conclusions, and will report both strengths and limitations of our prediction models. Correlations will not be interpreted as causal relationships.

C.4 Privacy in analysis: Have we ensured that data with PII are not used or displayed unless necessary for the analysis?

No personally identifiable information will be displayed or analyzed. Review text will be processed only for sentiment scoring and reported in aggregate form.

C.5 Auditability: Is the process of generating the analysis well documented and reproducible if we discover issues in the future?

All data processing and modeling steps will be documented in reproducible code to ensure transparency and allow for future review or correction if issues are identified.

D. Modeling

D.1 Proxy Discrimination: The primary risk here is using geographic location as a proxy for socioeconomic status or race. In Philadelphia, ZIP codes can reflect historical redlining or neighborhood wealth; if the model relies too heavily on location to predict "popularity," it may unfairly penalize businesses in underserved areas. We must ensure the model focuses on service-related features rather than location-based demographics that correlate with protected groups.

D.2 Disparate Impact: The model might achieve high accuracy overall but fail for specific cuisine types or niche markets. For instance, if the dataset is dominated by American fast food, the model’s "popularity predictors" might not accurately apply to authentic, family-owned ethnic restaurants in neighborhoods like Chinatown, South Philadelphia, or West Philadelphia. We should audit error rates across different categories to ensure the model isn't systematically "underrating" the potential of diverse businesses.

D.3 Metric Selection: Optimizing for average rating alone can be misleading, as a restaurant with five reviews is not necessarily more "popular" than one with 1,000 reviews and a 4-star average. We should use a weighted metric or a combination of review volume and sentiment density to capture a truer sense of popularity. This prevents the model from being skewed by small sample sizes or "review bombing" behavior.

D.5 Communicate Limitations: We must inform stakeholders that Yelp data is a "snapshot" and subject to selection bias, as not everyone leaves reviews. The model reflects the opinions of Yelp users—who may skew toward a specific demographic—rather than the entire Philadelphia population. Additionally, the model cannot account for external factors like sudden "TikTok trends" or road construction that might temporarily impact a restaurant’s foot traffic.

E. Deployment

E.1 & E.3 Monitoring and Rollback: Since popularity trends shift seasonally, we need to monitor the model for concept drift to ensure its predictions remain relevant. Because we are using Git for version control, we have a safety net: if a model update begins producing biased or nonsensical recommendations for certain Philadelphia neighborhoods, we can immediately revert to a previous, stable version of the code.

E.4 Unintended Use: While this tool is meant to help small business owners understand growth, it could be unintentionally used by real estate speculators or predatory lenders to identify areas for gentrification. We must clarify that the model is designed for operational improvement—helping a restaurant refine its menu or service—not as a tool to exclude or exploit businesses in "low-popularity" predicted zones.

## Team Expectations 

- Expect the team to communicate on Discord/Messages when people have questions.
- Expect to help fairly distribute tasks according to when we need to have them done.
- Expect the team to update one another when tasks are completed.
- Expect the team to fully understand the project's goal, and ask questions if there are any uncertainties.
- Expect individuals to complete tasks by deadlines, and communicate early if unable to do so.
- In the situation of a conflict: schedule a time to listen to both perspectives, discuss solutions, and proceed with the approach with the majority in agreement. 
- Always be nice and considerate.
- Expect teammates to offer help when others are having trouble.

## Project Timeline Proposal

| Meeting Date  | Meeting Time| Completed Before Meeting  | Discuss at Meeting |
|---|---|---|---|
| 2/3  |  4 PM | Think about group expectations expectations; brainstorm topics/questions  | Determine best form of communication; Discuss and decide on final project topic; Discuss hypothesis and topic; Revise and submit proposal | 
| 2/10  |  4 PM |  Brainstorm hypothesis and questions; Do background research on topic; Look for possible datasets; Confirm appropriate meeting times  | Discuss ethics; Confirm question and topic; Finalize dataset and discuss possible data wrangling and possible analytical approaches; Assign group members to lead each part of data wrangling and visualization | 
| 2/17  | 4 PM  | Finalize Background Section (Melanie); Import, Clean & Wrangle Data (Ethan, Katelyn, Angelina); EDA (Angelina)  |  Disucss Findings on Data; Review/Edit wranging/EDA; Finalize Analysis Plan; Complete Checkpoint   |
| 2/24  | 4 PM  | Begin Visualizations (Ethan), Begin Analysis (Melanie, Nimah)  | Discuss/edit Analysis Progress  |
| 3/2  | 4 PM  | Finalize wrangling/EDA (Katelyn, Angelina); Finish Most of the Analysis (Nimah); Finalize Visualizations (Ethan) | Discuss/edit Analysis; Complete project check-in |
| 3/10  | 4 PM  | Complete analysis (Nimah, Katelyn); Draft results/conclusion/discussion (Melanie)| Discuss/edit full project |
| 3/14  | Before 11:59 PM  | N/A | Turn in Final Project & Group Project Surveys |